app/dossierPrograms/dossierPrograms.controllers.js

```diff
@@ -100,19 +82,129 @@ dossierProgramsModule.controller("dossiersProgramSectionController", [
+                pss.dataElements.forEach(pssDE => {
+                    assignedDEArray.forEach(adeArray => {
+                        if (adeArray.ids.includes(pssDE.id)) {
+                            pssDE.calcMode = { type: "programRule", name: adeArray.name };
+                        } else if (!pssDE.calcMode) {
+                            pssDE.calcMode = { type: "default" };
+                        }
+                    });
+                });
+
+                if (hiddenSectionsArray.includes(pss.id)) {
+                    pss.dataElements.forEach(pssDE => {
+                        if (pssDE.calcMode.type !== "programRule") {
+                            pssDE.calcMode = { type: "other" };
+                        }
+                    });
+                }
+            });
+        }
+
+        /*
+         *  @name makeStageCalcMode
+         *  @description Determine the "Calculation mode" of the stage data elements
+         *  @scope dossiersProgramSectionController
+         */
+        function makeStageCalcMode(stage, assignedDEArray) {
+            stage.programStageDataElements.forEach(psde => {
+                assignedDEArray.forEach(adeArray => {
+                    if (adeArray.ids.includes(psde.dataElement.id)) {
+                        psde.dataElement.calcMode = { type: "programRule", name: adeArray.name };
+                    } else if (!psde.dataElement.calcMode) {
+                        psde.dataElement.calcMode = { type: "default" };
+                    }
+                });
+            });
+        }
+
+        /*
+         *  @name none
+         *  @description Gets the program stages information, translates it and shows it
+         *  @dependencies dossiersProgramStageSectionsFactory, dossiersProgramStageCalcModeFactory
+         *  @scope dossiersProgramSectionController
+         */
+        $scope.$watch("selectedProgram", function () {
+            ping();
+            if ($scope.selectedProgram) {
+                startLoadingState(false);
+                //Query sections and data elements
+                var stageSectionPromises = $scope.selectedProgram.programStages.map(function (stage) {
+                    return dossiersProgramStageSectionsFactory.get({ programStageId: stage.id }).$promise;
+                });
+
+                $q.resolve(
+                    dossiersProgramStageCalcModeFactory.get({ programId: $scope.selectedProgram.id }).$promise,
+                    data => ($scope.programRules = data.programRules)
+                ).then(() => {
+                    $q.all(stageSectionPromises).then(function (stages) {
+                        const hiddenSectionsArray = $scope.programRules.flatMap(pr => {
+                            return pr.programRuleActions.flatMap(pra => {
+                                if (pra.programRuleActionType === "HIDESECTION") {
+                                    return pra.programStageSection.id;
```

Comment:
[nitpick] Since it's a flatMap, it should return an array: `return [pra.programStageSection.id]`

Reply:
Done.